js/src/vm/JSScript.cpp

```diff
@@ -95,6 +95,186 @@ using JS::CompileOptions;
 using JS::ReadOnlyCompileOptions;
 using JS::SourceText;
 
+class FrontendAccessor {
+  using BytecodeEmitter = frontend::BytecodeEmitter;
+
+ public:
+  static inline void assertIncides(BytecodeEmitter* bce) {
+    /* The counts of indexed things must be checked during code generation. */
+    MOZ_ASSERT(bce->perScriptData().atomIndices()->count() <= INDEX_LIMIT);
```

Comment:
These assertions should remain in `JSScript::fullyInitFrom()`. Like this:

```
    MOZ_ASSERT(stencil.natoms <= INDEX_LIMIT);
    MOZ_ASSERT(stencil.ngcthings <= INDEX_LIMIT);
```

Reply:
(However, if you decide to keep the method, there's a typo; it should be `assertIndices`.)